plausible/events.go

```diff
@@ -0,0 +1,64 @@
+package plausible
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/valyala/fasthttp"
+)
+
+type Event struct {
+	baseURL    string
+	token      string
+	httpClient *fasthttp.Client
+}
+
+type Revenue struct {
+	Currency string `json:"currency"`
+	Amount   string `json:"amount"`
+}
+
+type EventRequest struct {
+	// the domain of your site (must match exactly)
+	Domain string `json:"domain"`
+
+	// the name of the event, e.g. 'pageview', otherwise a custom event
+	Name string `json:"name"`
+
+	// the URL of the current request
+	URL *url.URL `json:"url"`
+
+	// optional parameters
+	Referrer string            `json:"referrer,omitempty"`
+	Props    map[string]string `json:"props,omitempty"`
+	Revenue  Revenue           `json:"revenue,omitempty"`
+}
+
+func (e *Event) Push(data EventRequest, headers map[string]string, event string) ([]byte, error) {
+	userAgent, ok := headers["user-agent"]
+	if !ok {
+		return nil, fmt.Errorf("missing user-agent header")
+	}
+
+	req := fasthttp.AcquireRequest()
+	req.SetRequestURI(e.baseURL + "/api/event")
+	req.Header.SetMethod(fasthttp.MethodPost)
+	req.Header.Add("Authorization", "Bearer "+e.token)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("User-Agent", userAgent)
+
+	if forwardedFor, ok := headers["x-forwarded-for"]; ok {
+		req.Header.Add("X-Forwarded-For", forwardedFor)
+	}
```

Comment:
The logic to acquire an authenticated request is already present in [here](https://github.com/andrerfcsantos/go-plausible/blob/main/plausible/client.go#L84). I think we should reuse the `(c *Client) acquireRequest` method here to acquire an authenticated request to an endpoint. In another comment I made the suggestion of making `Push` be a method on the `c *Client` directly or make `Event` take the `*Client` directly. Both options would help in calling the `(c *Client) acquireRequest` trivial here. 

The logic to add the additional headers and encode the body as json can stay here.